### ISSUE TITLE
K3s fails on Alpine flavors

### ISSUE THREAD

**Issue by christianprim** [kairos-io/kairos#484]
**Kairos version:**  v1.2.0

**CPU architecture, OS, and Version:**  amd64, alpine-ubuntu and alpine-opensuse-leap

**Describe the bug**

K3s fails to start on Alpine flavors. Maybe a problem with cgroups?

**To Reproduce**

Interactive install with k3s enabled (or k3s-agent enabled). Reboot. K3s fails to start.

**Expected behavior**

K3s should start automatically.

**Logs**

Maybe not helpful:

```
sudo rc-service k3s (or k3s-agent) status
* status: stopped

sudo rc-service k3s (or k3s-agent) start
... level=fatal msg="failed to find cpu cgroup (v2)"
```

And for the k3s-agent:
```
cat /var/log/kairos/agent.log
Provider kairos at /system/providers/agent-provider-kairos had an error: Failed setup: exit status 127
```

**Comment by mudler** [comment]
Wondering if it was caused by https://github.com/kairos-io/kairos/issues/491 ?

**Comment by christianprim** [comment]
A possible fix:
```
stages:
   initramfs:
     - name: "Fix rc.conf"
       commands:
       - echo 'rc_cgroup_mode="hybrid"' >> /etc/rc.conf
       - echo 'rc_cgroup_controllers="cpuset cpu io memory hugetlb pids"' >> /etc/rc.conf
```

**Comment by christianprim** [comment]
Above fix doesn't solve the initial issue. k3s-agent still doesn't start automatically. There must be an issue with kairos-agent. Copying the env variables in /etc/rancher/k3s/k3s-agent.env and starting the k3s-agent manually works with the fix above.

Same for k3s: one must start the service manually. It doesn't start automatically. So the initial issue remains.

**Comment by mudler** [comment]
thanks for confirming! will go along the lines of your solution for now. I think those settings can be also set directly in `/etc/conf.d/`, however `rc.conf` looks also fine as well

**Comment by mudler** [comment]
I believe `k3s` doesn't start as there is no `#cloud-config` header in your files @christianprim . can you confirm?

https://github.com/kairos-io/kairos/pull/502 allows merging all config founds greedly - as soon as it's merged you can expect it in the next releases

**Comment by christianprim** [comment]
I have the header:

```
#node-config
install:
  device: /dev/sda
k3s:
  enabled: true
name: Config generated by the installer
stages:
  initramfs:
  - commands:
    - echo 'rc_cgroup_mode="hybrid"' >> /etc/rc.conf
    - echo 'rc_cgroup_controllers="cpuset cpu io memory hugetlb pids"' >> /etc/rc.conf
    - mount --make-rshared /
  network:
  - users:
      kairos:
        groups:
        - admin
        name: kairos
        passwd: kairos
        ssh_authorized_keys:
        - ""
 ```
 I used the interactive install and added the few lines for k3s to work. So that can't be the issue.

**Comment by christianprim** [comment]
But I get:

```
# cat /var/log/kairos/agent.log
Provider kairos at /system/providers/agent-provider-kairos had an error: Failed setup: exit status 127
```

**Comment by christianprim** [comment]
Reading the code: The issue must be in the oneTimeBootstrap-function between L177 (https://github.com/kairos-io/provider-kairos/blob/main/internal/provider/bootstrap.go#L177) and L197 since the envFile isn't written when role is set to agent, but the info (One time bootstrap starting) is logged to /var/log/kairos/agent.log.

Hope it helps to narrow down the issue.

**Comment by christianprim** [comment]
fixed in v1.3.0
